pandas/tseries/frequencies.py

```diff
@@ -13,18 +13,23 @@
     is_timedelta64_dtype,
     is_datetime64_dtype)
 
-import pandas.core.algorithms as algos
-from pandas.core.algorithms import unique
 from pandas.tseries.offsets import DateOffset
-from pandas.util._decorators import cache_readonly, deprecate_kwarg
+from pandas.util._decorators import deprecate_kwarg
 import pandas.tseries.offsets as offsets
 
-from pandas._libs import lib, tslib
+from pandas._libs import tslib
 from pandas._libs.tslib import Timedelta
 from pandas._libs.tslibs.frequencies import (  # noqa
     get_freq_code, _base_and_stride, _period_str_to_code,
     _INVALID_FREQ_ERROR, opattern, _lite_rule_alias, _dont_uppercase,
     _period_code_map, _reverse_period_code_map)
+from pandas._libs.tslibs.resolution import (  # noqa
```

Comment:
why the noqa?

Reply:
It's public-facing so I didn't want to break anything downstream.  Remove unused imports?

Reply:
hmm, what exactly is public facing of these? I am ok with simply removing anything _leading, what is not here?

Reply:
I think the noqa is for `DAYS`, `_maybe_add_count`, `_weekday_rule_aliases`, `get_freq_group`.  I think the private ones here can be removed without breaking anything.  `DAYS` and `get_freq_group` are imported by and a couple other places.

Reply:
ok let's remove any non-used, you can put the non-private ones in a whatsnew note

Reply:
Just pushed.  Wasn't sure about wording in whatsnew note for the list `DAYS`.